src/DynamoPackages/PackageManagerExtension.cs

```diff
@@ -180,6 +225,7 @@ public void Ready(ReadyParams sp)
 
         public void Shutdown()
         {
+            PythonServices.PythonEngineManager.Instance.AvailableEngines.CollectionChanged -= PythonEngineAdded;
```

Comment:
have you verified this actually gets called? I cannot remember if it's this or dispose for extensions.

Reply:
yes, verified.